### PR TITLE
Update Blacklight::SearchState to play well with Rails 5 AC::Parameters class

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: lib/blacklight/search_state.rb [hunk 1]
```diff
@@ -10,22 +10,26 @@ class SearchState
...
-        # https://github.com/rails/rails/pull/22913 (Rails > 4.2.5)
-        @params = params.with_indifferent_access
-      else
+      if params.respond_to?(:to_unsafe_h)
         # This is the typical (not-ActionView::TestCase) code path.
         @params = params.to_unsafe_h
         # In Rails 5 to_unsafe_h returns a HashWithIndifferentAccess, in Rails 4 it returns Hash
         @params = @params.with_indifferent_access if @params.instance_of? Hash
+      elsif params.is_a? Hash
+        # This is an ActionView::TestCase workaround. Will be resolved by
```
Comment (jcoyne) [inline lib/blacklight/search_state.rb]: Will not be resolve apparently. Rafael declined to merge it.

Comment (cbeer) [inline lib/blacklight/search_state.rb]: :/ .. I'll remove the comment too.

File: lib/blacklight/search_state.rb [hunk 2]
```diff
@@ -10,22 +10,24 @@ class SearchState
...
-        # https://github.com/rails/rails/pull/22913 (Rails > 4.2.5)
-        @params = params.with_indifferent_access
-      else
+      if params.respond_to?(:to_unsafe_h)
         # This is the typical (not-ActionView::TestCase) code path.
         @params = params.to_unsafe_h
         # In Rails 5 to_unsafe_h returns a HashWithIndifferentAccess, in Rails 4 it returns Hash
         @params = @params.with_indifferent_access if @params.instance_of? Hash
+      elsif params.is_a? Hash
+        @params = params.dup.with_indifferent_access
```
Comment (jcoyne) [inline lib/blacklight/search_state.rb]: Can you add a comment here that this is a workaround for AV::TestCase and is only needed for rails 4.2 (so we can remove it down the road?)

Comment (cbeer) [inline lib/blacklight/search_state.rb]: 👌

**Pull request merged**